File: Array (MS)/1_Two_Sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  var hashmap = new Map();

  for (let i = 0; i < nums.length; i++) {
    if (hashmap.has(target - nums[i])) {
      return [hashmap.get(target - nums[i]), i];
    }

    hashmap.set(nums[i], i);
  }
};

// Niceee solution

File: Array (MS)/1_Two_Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        record = {}
        for i in range(len(nums)):
            supplement = target - nums[i]
            if supplement in record:
                return [record[supplement], i]
            else:
                record[nums[i]] = i

        return []
